src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -608,6 +608,62 @@ public void RegisterSymbolAction(DiagnosticAnalyzer analyzer, Action<SymbolAnaly
             SymbolAnalyzerAction analyzerAction = new SymbolAnalyzerAction(action, symbolKinds, analyzer);
             this.GetOrCreateAnalyzerActions(analyzer).AddSymbolAction(analyzerAction);
             _symbolActions = _symbolActions.Add(analyzerAction);
+
+            // The SymbolAnalyzerAction does not handle SymbolKind.Local or SymbolKind.Parameter
+            // because the compiler does not make CompilationEvents for them. As a workaround,
+            // handle them specially by registering further OperationActions (for 
+            // VariableDeclarations) or SymbolActions (for Methods) and utilize the results of
+            // those to construct the necessary SymbolAnalysisContexts.
+
+            if (symbolKinds.Contains(SymbolKind.Local))
+            {
+                RegisterOperationAction(analyzer, opContext =>
+                {
+                    var varDecl = (IVariableDeclaration)opContext.Operation;
+                    action(new SymbolAnalysisContext(
+                        varDecl.Variable,
+                        opContext.Compilation,
+                        opContext.Options,
+                        opContext.ReportDiagnostic,
+                        opContext.IsSupportedDiagnostic,
+                        opContext.CancellationToken));
+                },
+                ImmutableArray.Create(OperationKind.VariableDeclaration));
+            }
+
+            if (symbolKinds.Contains(SymbolKind.Parameter))
+            {
+                RegisterSymbolAction(
```

Comment:
Right approach would be to avoid registering a separate action here, and instead just iterate over all parameters of the symbol [here](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalyzerExecutor.cs,325) and invoke symbol actions for parameters.